dask_xgboost/core.py

```diff
@@ -172,32 +175,37 @@ def _train(
     if isinstance(label_parts, np.ndarray):
         assert label_parts.ndim == 1 or label_parts.shape[1] == 1
         label_parts = label_parts.flatten().tolist()
+    if sample_weight is not None:
+        sample_weight_parts = sample_weight.to_delayed()
+        if isinstance(sample_weight_parts, np.ndarray):
+            assert sample_weight_parts.ndim == 1 or sample_weight_parts.shape[1] == 1
+            sample_weight_parts = sample_weight_parts.flatten().tolist()
+    else:
+        # If sample_weight is None construct a list of Nones to keep
+        # the structure of parts consistent.
+        sample_weight_parts = [None] * len(data_parts)
 
-    # Arrange parts into pairs.  This enforces co-locality
-    parts = list(map(delayed, zip(data_parts, label_parts)))
+    # Arrange parts into triads.  This enforces co-locality
+    parts = list(map(delayed, zip(data_parts, label_parts, sample_weight_parts)))
```

Comment:
Hmmm, we should be verifying that these are the same length, otherwise the `zip` will silently drop data. Oh well, it's a problem on master too :)

Reply:
That's a good idea. I can add a check.